### PR TITLE
feat: implementing sso orchestrator existing configs page

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
https://2u-internal.atlassian.net/browse/ENT-7579

information on rendering elements based on states of configs: https://docs.google.com/spreadsheets/d/1DKIkXohDWJ3OkuMNAkvoG_9RGjSD_SAA_5cU7oBQ9j4/edit#gid=0 

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    },
+    // Refetch the data every second
+    refetchInterval: intervalMs,
+    enabled: queryForTestedConfigs || queryForConfiguredConfigs,
+    refetchOnWindowFocus: true,
+  });
+
+  useEffect(() => {
+    const inactive = configs.filter(config => config.active === false);
+    const active = configs.filter(config => config.active === true);
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx:71]: nit: Could simplify a bit using [lodash.partition](https://lodash.com/docs/4.17.15#partition)
```suggestion
    const [active, inactive] = _.partition(configs, config => config.active);
```

File: src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  const [queryForTestedConfigs, setQueryForTestedConfigs] = useState(false);
+  const [queryForConfiguredConfigs, setQueryForConfiguredConfigs] = useState(false);
+  const [intervalMs, setIntervalMs] = React.useState(FRESH_CONFIG_POLLING_INTERVAL);
+  const [loading, setLoading] = useState(false);
+  const [showAlerts, openAlerts, closeAlerts] = useToggle(false);
+
+  const queryClient = useQueryClient();
+
+  useQuery({
+    queryKey: ['todos'],
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx:35]: I see the queryKey `todos` in [tanstack sample code that uses Todo objects](https://tanstack.com/query/v4/docs/react/guides/query-keys), but should we be using something different here?

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 3]
```diff
@@ -21,39 +24,115 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
   const [existingProviderData, pdError, pdIsLoading] = useExistingProviderData(enterpriseId, refreshBool);
   const [showNewSSOForm, setShowNewSSOForm] = useState(false);
   const [showNoSSOCard, setShowNoSSOCard] = useState(false);
+  const { AUTH0_SELF_SERVICE_INTEGRATION } = features;
+  const [isOpen, open, close] = useToggle(false);
+
+  const newConfigurationButtonOnClick = async () => {
+    for (let i = 0; i < existingConfigs.length; i++) {
+      LmsApiService.updateEnterpriseSsoOrchestrationRecord(
```
[the remark this answers is not in the document]
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/index.jsx:32]: good idea

File: src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx (hunk 2 above)
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx:35]: yeah happy to make it something more related to our project, figured it was a key thing per project so I wasn't that worried but happy to add our own flair 👍

File: src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+          {showAlerts && (
+            <NewSSOConfigAlerts
+              liveConfigs={liveConfigs}
+              inProgressConfigs={inProgressConfigs}
+              untestedConfigs={untestedConfigs}
+              notConfigured={notConfiguredConfigs}
+              closeAlerts={closeAlerts}
+            />
+          )}
+          {activeConfigs.length > 0 && (
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/NewExistingSSOConfigs.jsx:123]: nit: activeConfigs and inactiveConfig sections are identical aside from header text and config list, might be worth making a helper function for readability.

File: src/components/settings/SettingsSSOTab/NewSSOConfigCard.jsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      )}
+      {config.validated_at && config.submitted_at < config.configured_at && !config.active && (
+        <Badge className="ml-4 p-2 font-weight-light" variant="light">Disabled</Badge>
+      )}
+    </>
+  );
+
+  const renderCardButton = () => (
+    <>
+      {!config.validated_at && (config.submitted_at < config.configured_at) && (
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/NewSSOConfigCard.jsx:94]: nit: We're doing a lot of embedded state checks on the config in question.  I'm wondering if the logic would be easier to follow if we did a check up front to determine the state of the config (VALIDATED, DISABLED, ENABLED) and then base all the conditional renderings off that.

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 6]
```diff
@@ -21,22 +24,143 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
   const [showNoSSOCard, setShowNoSSOCard] = useState(false);
+  const { AUTH0_SELF_SERVICE_INTEGRATION } = features;
+  const [isOpen, open, close] = useToggle(false);
+
+  const newConfigurationButtonOnClick = async () => {
+    for (let i = 0; i < existingConfigs.length; i++) {
+      Promise.all(LmsApiService.updateEnterpriseSsoOrchestrationRecord(
+        { active: false, is_removed: true },
+        existingConfigs[i].uuid,
+      )).then(() => {
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/index.jsx:35]: Make sure we're calling Promise.all() on all the calls, not each one individually.
```suggestion
      Promise.all(existingConfigs.map(config => LmsApiService.updateEnterpriseSsoOrchestrationRecord(
        { active: false, is_removed: true },
        config.uuid,
      ))).then(() => {
```

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 7]
```diff
@@ -21,22 +24,143 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
   useEffect(() => {
-    let validConfigExists = false;
-    existingConfigs.forEach(config => {
-      if (config.was_valid_at) {
-        validConfigExists = true;
-      }
-    });
+    if (AUTH0_SELF_SERVICE_INTEGRATION) {
+      let validConfigExists = false;
+      validConfigExists = existingConfigs.some(config => config.validated_at);
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/index.jsx:45]:
```suggestion
      let validConfigExists = existingConfigs.some(config => config.validated_at);
```

File: src/components/settings/SettingsSSOTab/index.jsx [hunk 8]
```diff
@@ -21,22 +24,143 @@ const SettingsSSOTab = ({ enterpriseId, setHasSSOConfig }) => {
...
+      setHasSSOConfig(validConfigExists);
+    } else {
+      let validConfigExists = false;
+      existingConfigs.forEach(config => {
+        if (config.was_valid_at) {
+          validConfigExists = true;
+        }
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/index.jsx:52]: Can condense this logic with array.some() like the block above

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/components/settings/SettingsSSOTab/index.jsx (hunk 8 above)
Comment (alex-sheehan-edx) [reply src/components/settings/SettingsSSOTab/index.jsx:52]: yeah I considered it, just didn't want to touch existing implementations much but I guess it's best practice

File: src/components/settings/SettingsSSOTab/NewSSOConfigCard.jsx [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+const NewSSOConfigCard = ({
+  config,
+  setLoading,
+  setRefreshBool,
+  refreshBool,
+}) => {
+  const VALIDATED = config.validated_at;
+  const ENABLED = config.active;
+  const CONFIGURED = config.submitted_at < config.configured_at && config.configured_at;
```
Comment (marlonkeating) [inline src/components/settings/SettingsSSOTab/NewSSOConfigCard.jsx:20]: nit: LTR logic and parentheses for readability.
```suggestion
  const CONFIGURED = config.configured_at && (config.submitted_at < config.configured_at);
```

**Pull request merged**